packages/keyring-controller/src/KeyringController.ts

```diff
@@ -680,17 +734,24 @@ export class KeyringController extends BaseControllerV2<
       throw new Error('No HD keyring found.');
     }
 
+    assertHasUint8ArrayMnemonic(primaryKeyring);
+
     const seedWords = primaryKeyring.mnemonic;
     const accounts = await primaryKeyring.getAccounts();
     /* istanbul ignore if */
     if (accounts.length === 0) {
       throw new Error('Cannot verify an empty keyring.');
     }
 
+    // The HD Keyring Builder is a default keyring builder
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
     const hdKeyringBuilder = this.#keyring.getKeyringBuilderForType(
       KeyringTypes.hd,
-    );
+    )!;
+
     const hdKeyring = hdKeyringBuilder();
+    // @ts-expect-error @metamask/eth-hd-keyring correctly handles
```

Comment:
This typecast seems like a problem with the Keyring interface itself: it expects a state object, but it's getting serialized state, which has a different shape. This is a type error; we shouldn't be using the same type for serialized and unserialized state.

Reply:
Do you think we can leave this as is for now?